symex-primitives.el

```diff
@@ -377,6 +385,8 @@ of symex mode (use the public `symex-go-up` instead)."
              (cond ((looking-back "#['`]" (line-beginning-position))
                     (backward-char 2))
                    ((looking-back "[#'`]" (line-beginning-position))
+                    (backward-char))
+                   ((looking-back "@" (line-beginning-position))
```

Comment:
"Looking back" at this code, I'm wondering whether we could reuse the predicates defined earlier instead of checking for the pattern independently here. Maybe it would entail moving the point back until it reaches the first non-whitespace character or bol (beginning of line), and then checking the relevant predicates e.g. `(clojure-deref-reader-macro-p)`. This is outside the scope of this PR though (could be a separate PR if you are interested in taking a crack at it), just making a note of it.